src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -1134,6 +1142,7 @@ infectiveCoinfectiveOptions =
                                               "--no-universe-polymorphism"
   , coinfectiveOption (not . optCumulativity) "--no-cumulativity"
   , coinfectiveOption optLevelUniverse        "--level-universe"
+  , coinfectiveOption (not . optEtaSingleton) "--no-eta-singleton"
```

Comment:
Shouldn't `--eta-singleton` be coinfective? I thought the idea was to make sure that no types are "problematic", and in that case it should not be possible to import problematic types when `--eta-singleton` is used.